### PR TITLE
GridView PR 1

### PR DESCRIPTION
**Pull request by yangdanny97**
The grid lines for GridView are implemented. 

- Zooming/Panning is currently implemented with three sliders for development purposes, but this is subject to change later. 
- Some params for functions have been changed (specifically passing in x+y offsets and scale is no longer necessary since it's stored in the state)
- Split initial setup of gridview into two parts, main() and setup() to better handle background image loading (error handling doesn't work here for some reason, so I disabled it and made it default to a white background)

Incomplete things:
- some bot/vision related things aren't implemented due to lack of vision system
- same for occupancy matrix related things
- ajax requests for the above need to be converted to axios
- as mentioned above background image loading is disabled for now
- keyboard controls for panning are not implemented

### CODE REVIEW

**Comment by leotrj** [comment]
How was this tested? 
Can we see some screenshots?

**Comment by yangdanny97** [comment]
Manual testing (just jump to that branch and open up index, the grid will display even when offline).

File: gui/main.py [hunk 1]
```diff
@@ -7,8 +7,10 @@
 import os.path
 import json
 import logging
+import sys
 
 # Minibot imports.
+sys.path.insert(0, basestation)
```
Comment (chynu) [inline gui/main.py]: Why do we have this here again?

**Review by chynu — COMMENTED** [review]
Not sure why, but when I run the GUI, I get an error:

image

I have a feeling this has to do with your line that inserts `basestation` at the top of `main.py`. Did you run into a similar issue? This doesn't happen on `develop`.

File: gui/main.py (hunk 1 above)
Comment (yangdanny97) [reply gui/main.py]: hm i think that line is unnecessary; I added it when basestation wouldn't run on my computer due to some path issue

**Comment by chynu** [comment]
I don't see anything when I run the GUI? And the DOM doesn't have anything inside the gridview div. Is this just me, am I testing it incorrectly? :o

File: gui/static/js/components/gridview.js [hunk 2]
```diff
@@ -1,17 +1,628 @@
...
+                    scenarioObject.x = cx;
+                    scenarioObject.y = cy;
+                    botContainer.addChild(scenarioObject);
+                }
+            }
+        }
+    }
+
+
+    //TODO - keyboard control for panning - DO WE WANT TO KEEP THIS?
```
Comment (chynu) [inline gui/static/js/components/gridview.js]: Yes, it's a good idea imo.

File: gui/static/js/components/gridview.js [hunk 3]
```diff
@@ -1,17 +1,628 @@
...
+                        <td className = "gridControlTable"><input id="yOffset" type="range" name="yOffset" min="-100" max="100" className = "gridControl" value={this.state.yOffset} defaultValue="0" onChange={this.handleInputChange}/></td>
+                    </tr>
+                </tbody></table>
+                <div id ="view"></div>
+            </div>
+        );
     }
 }
+
+
```
Comment (chynu) [inline gui/static/js/components/gridview.js]: Remove unnecessary newlines.

File: gui/static/js/components/gridview.js [hunk 4]
```diff
@@ -1,17 +1,628 @@
...
+        var bot = {
+            x: x,
+            y: y,
+            angle: angle,
+            id: id,
+            size: size
+        };
+        /*in the future add something to identify bots vs objects*/
+        if (size==0.15){
+            bot.type = 'bot';}
```
Comment (chynu) [inline gui/static/js/components/gridview.js]: Nit: funky formatting?

**Review by chynu — CHANGES_REQUESTED** [review]
I'll address them, just commenting for the record.

**Review by chynu — APPROVED** [review]

**Pull request merged**